### We try to explore some basic relationships from the trec relevant reports. We also investigate where my method does well and fails. 


Basic todo:
* parse relevance judgement score for each topic -- done
* get a document by quering from the sql database given a document ID -- done
* spot patterns in good scoring documents
* 

In [1]:
import mysql.connector
import pandas as pd
import matplotlib.pyplot as plt

mydb = mysql.connector.connect(
  host="localhost",
  user="root",
  passwd="root",
  database="wapo"
)

mycursor = mydb.cursor()

mycursor.execute("SHOW TABLES")

for x in mycursor:
  print(x)

(u'reports',)


In [2]:
question_file_str = "/home/i3/git/WapoDataMining/resources/topics.xml"
output_dir_str = "/home/i3/git/WapoDataMining/results/"
answer_file_str = "/home/i3/git/WapoDataMining/bqrels.exp-gains.txt"
#rep_id = "00f57310e5c8ec7833d6756ba637332e"

In [3]:

from xml.dom import minidom
    
def getText(nodelist):
    rc = []
    for node in nodelist:
        if node.nodeType == node.TEXT_NODE:
            rc.append(node.data)
    return ''.join(rc)

def read_all_questions():
    """
     read the question xml file
     return a list of tuples
     [(topic_no,report_id),...]
    """

    newsitems = [] 
    
    xmldoc = minidom.parse(question_file_str)
    
    itemlist = xmldoc.getElementsByTagName('top')
    #print(len(itemlist))
    #print(itemlist[0].attributes['name'].value)
    for s in itemlist:
        
        #print(s.attributes['num'].value)
        docid = (getText(s.getElementsByTagName('docid')[0].childNodes))
        num = (getText(s.getElementsByTagName('num')[0].childNodes))
        num = num[9:len(num)-1]
        newsitems.append((num,docid))
        
    return newsitems
    
    
questions = read_all_questions()
#questions

"""
returns list of tuple
[(topic_id,doc_id,score), ...]
"""
def read_all_judgements():
    results = []
    f = open(answer_file_str, "r")
    for x in f:
        d = x.split()
        results.append((d[0],d[2],d[3]))
    f.close()
    return results
    
answers = read_all_judgements()
answers

[('321', '00f57310e5c8ec7833d6756ba637332e', '16'),
 ('321', '02ae7136-006d-11e3-9711-3708310f6f4d', '2'),
 ('321', '035622671d5ee6a098d686c9a4d5b18e', '16'),
 ('321', '041e530b21766fc294d4e9c48b1e2723', '0'),
 ('321', '049739753ce2e539d8dc2165daaf6aa6', '4'),
 ('321', '071068774e4eb454c1702192fcca886b', '2'),
 ('321', '071ad6f3cfba2273c7f8d5264c0a7583', '16'),
 ('321', '07578ec30a042be5678bda5f99949ff3', '0'),
 ('321', '090524c518ebedafae7ed68e16a25436', '0'),
 ('321', '09b3167f0d1aa5cfa8be932bb704d75a', '8'),
 ('321', '0a378bae-7833-11e1-b191-2d5f686e7ff4', '8'),
 ('321', '0bca80bf2a80b62a0e216540861ef7a9', '2'),
 ('321', '0caa33d172c696cac4bbc36df4e44bf2', '8'),
 ('321', '0da8ee91889db59cd8d8c4437655a0ef', '16'),
 ('321', '0e92c3bb3dbf2dc654a2a854752651ce', '8'),
 ('321', '1114e343-d128-4a7d-bb5d-66d2a9836f72', '2'),
 ('321', '12089ed7089574776db6c1e5e55704af', '8'),
 ('321', '13381e831a417a6db347ec237373d3ff', '4'),
 ('321', '139c0f1649fa758d2379555c91b9c37e', '4'),
 ('321', '13f8a

Now questions and answers contains the file contents. make them into dataframe.

In [4]:
answers_tup_df = pd.DataFrame(answers,columns=["topic_no","id","score"])
answers_tup_df

questions_tup_df = pd.DataFrame(questions,columns=["topic_no","id"])
questions_tup_df
print(answers_tup_df.shape)
print(questions_tup_df.shape)

(8508, 3)
(50, 2)


In [5]:

buf = "("
l = len(questions)

for q in questions:
    buf+="'"+q[1]+"'"
    l = l-1
    if(l!=0):
        buf+=","

buf+=")"

buf
query = "SELECT * FROM reports WHERE id IN "+buf
df_questions = pd.read_sql(query, con=mydb)
df_questions


id  \
0       02e52bdba097c9df4cbae66e04f82542   
1   0e85b0c0-f7ef-11e4-9030-b4732caefe81   
2   11915bd8-7944-11e2-9c27-fdd594ea6286   
3       145b9a6caa16d931c108a89798e65e17   
4   159e6f9e-8e84-11e3-84e1-27626c5ef5fb   
5       1e03fecf4d33b7896203298ab3858156   
6       2a340b8573d498e261d6f2365b37f8eb   
7       2bea9433d4e1050c9c85175df466b3e2   
8       2e83ad87eb1bade22e6e96ece616c24f   
9   30a493b8-fb07-11e4-9ef4-1bb7ce3b3fb7   
10  30c00b60-13f6-11e3-b182-1b3bb2eb474c   
11      37a8e2283e4677b703f6464d0191a700   
12      3902c9005a0563742fc4acb2c011b164   
13  3c5be31e-24ab-11e5-b621-b55e495e9b78   
14  4192b016-8708-11e3-a5bd-844629433ba3   
15      43e9f3f12982c0e0bb15ad64b33a89c0   
16  474ae088-ab1e-11e4-9c91-e9d2f9fde644   
17      4989ebfeb752e6b317d1ef3997b21a01   
18  563fb77e-024f-11e6-9203-7b8670959b88   
19      56f0438ee0fb34c341ccf5af36de5175   
20  579e9ae8-6a2f-11e6-8225-fbb8a6fc65bc   
21      5c466d4a01492f1b5cc9758e19429a1f   
22      5ec40b6bc6c5f4487132da7be04fc914   
23      5f37aac53768e749b861028397eb6849   
24      6668d83480f5c58b54a90770835ac2d4   
25      6fdc62d37aaf685b809c501abe13c56c   
26      72e72b41097d53b627fd375dd2d3309b   
27      7ef8ce1720bf2f6b2065a97506ee89b4   
28      9171debc316e5e2782e0d2404ca7d09d   
29  985b90cc-7c98-11e3-93c1-0e888170b723   
30      988147454a2b8eafd1535cd673dd04ba   
31      9dd7b85cd1e3da1b5c8e79f32fec7177   
32  a1c41a70-35c7-11e3-8a0e-4e2cf80831fc   
33      a244d1e0cfd916a2af76b6a6c785b017   
34      a2744bb98e1968307548e4976232cf1c   
35  a36fa8a2-8962-11e6-bff0-d53f592f176e   
36      a79b1b7d8cc5273d4995fec5e122e44b   
37  b0235f56-1cce-11e4-ae54-0cfe1f974f8a   
38      b4c6361974466458bb721b9b1628220b   
39      bd1e6cc8d7525fec36a717be45638bf4   
40      c109cc839f2d2414251471c48ae5515c   
41      c3cea789141ef2ae856419e86e165e0c   
42      c6bf4a4bf542b7c67987c222d73def4b   
43  c8351276-76de-41f1-b294-4f3e5d373c8c   
44      cad56e871cd0bca6cc77e97ffe246258   
45      dcd1560bd13a0b665b95d3ba27cc960c   
46      defd7f4a85496d52a210938d58a7ae76   
47  e1336b8f-b0c2-4610-9a3c-ec85a546c9ad   
48      fc1ca759c9c433376e71884870d225ab   
49      fef0f232a9bd94bdb96bac48c7705503   

                                                  url  \
0   https://www.washingtonpost.com/news/speaking-o...   
1   https://www.washingtonpost.com/business/econom...   
2                                           duplicate   
3   https://www.washingtonpost.com/news/post-natio...   
4   https://www.washingtonpost.com/entertainment/m...   
5   https://www.washingtonpost.com/news/worldviews...   
6   https://www.washingtonpost.com/news/animalia/w...   
7   https://www.washingtonpost.com/news/morning-mi...   
8   https://www.washingtonpost.com/news/capital-we...   
9   https://www.washingtonpost.com/world/national-...   
10  https://www.washingtonpost.com/lifestyle/style...   
11  https://www.washingtonpost.com/news/wonk/wp/20...   
12  https://www.washingtonpost.com/news/worldviews...   
13  https://www.washingtonpost.com/world/europe/to...   
14  https://www.washingtonpost.com/politics/sugar-...   
15  https://www.washingtonpost.com/blogs/erik-wemp...   
16  https://www.washingtonpost.com/local/traffican...   
17  https://www.washingtonpost.com/news/post-natio...   
18  https://www.washingtonpost.com/local/traffican...   
19  https://www.washingtonpost.com/news/wonk/wp/20...   
20  https://www.washingtonpost.com/local/md-politi...   
21  https://www.washingtonpost.com/news/speaking-o...   
22  https://www.washingtonpost.com/news/capital-we...   
23  https://www.washingtonpost.com/news/wonk/wp/20...   
24  https://www.washingtonpost.com/news/worldviews...   
25  https://www.washingtonpost.com/blogs/govbeat/w...   
26  https://www.washingtonpost.com/news/energy-env...   
27  https://www.washingtonpost.com/news/dr-gridloc...   
28  https://www.washingtonpost.com/news/worldviews...   
29  https://www.washingtonpost.com/national/health...   
30  https://www.washington

In [6]:
buf = "("
l = len(answers)

for q in answers:
    buf+="'"+q[1]+"'"
    l = l-1
    if(l!=0):
        buf+=","

buf+=")"

query = "SELECT * FROM reports WHERE id IN "+buf
df_answers = pd.read_sql(query, con=mydb)
df_answers

id  \
0         00041887f95b15333ae39503610f2de1   
1     00054d36-2473-11e3-b3e9-d97fb087acd6   
2         000563252f7569174d0f1a2aff115854   
3     000da51c-f93f-11e2-a954-358d90d5d72d   
4         000f4c71d82856a1584c4a3c62c71454   
5         0013cc4281f4c81f611bb625031ac605   
6     0015851c-2170-11e3-966c-9c4293c47ebe   
7         003014d2b6c491c98fa759692e0c99da   
8         0032b35853f8288d99d4f92cb5794966   
9         00461b9f108425ff68f1dd910a6cf202   
10        005308753e2d6f8a71c8be31be210e28   
11        0058b29067c414f21b35c8f64cb49e64   
12        005a1f0c2064487a7f15443b2a5f349a   
13    0062ad00-f7ee-11e4-a47c-e56f4db884ed   
14        006f205a6298a2ff627b5236cb984463   
15        007506cd7a7fcc1e03aac3a8e93ba6db   
16        00796cc1f45678e59898bee51e7260ae   
17        007a0486368dd6aa54f2b62cd431c046   
18        007cf8e2f0867bf2db10a49162386e83   
19    007f0b42-0e28-11e2-bd1a-b868e65d57eb   
20        008026efacc89ec065782505f4da4472   
21        00809043cc3bbea3b4d584ce62d08b0b   
22        00876e25a722ae351db811349a8bd101   
23    0089c108-0d53-11e6-bfa1-4efa856caf2a   
24    008c981a-c2e0-11e4-ad5c-3b8ce89f1b89   
25        0091c37f78cae2ab67cff31a52eb4f1f   
26    0091e9d4-c8f1-11e4-a199-6cb5e63819d2   
27    009aafb6-0283-11e6-8bb1-f124a43f84dc   
28        00a43ad2b87158552fa26eb0c9041df6   
29        00b591984edb37b8b99d3ef266e6bbfe   
...                                    ...   
8245      ff0a760128ecdbcc096cafc8cd553255   
8246  ff0ae020-4a6e-11e5-8ab4-c73967a143d3   
8247  ff13bb5c-aea5-11e3-a49e-76adc9210f19   
8248  ff1f5c2c-10ab-11e6-8967-7ac733c56f12   
8249  ff2d3122-21f2-11e2-bdfa-eebc58545bc7   
8250  ff308aa4-822f-11e3-bbe5-6a2a3141e3a9   
8251  ff3a25b0-0ba4-11e4-8341-b8072b1e7348   
8252      ff42f6cd0547632631044913241fec0d   
8253  ff47b876-714d-11e2-ac36-3d8d9dcaa2e2   
8254  ff49e708-5771-11e3-ba82-16ed03681809   
8255      ff4c7b7a86185b580010f20fad44a6d5   
8256      ff52231aec12758f729eeb4c56fc902a   
8257      ff56952e96544c44067ffc403fc753ba   
8258      ff59f055c6f05e9ce3aa91461f492bc2   
8259      ff63aded5d71af8cc553f22dca202370   
8260  ff6bd6e0-fe0e-11e2-9711-3708310f6f4d   
8261  ff7a152a-7303-11e7-9eac-d56bd5568db8   
8262      ff871ed540102bbc7ff03aba759f38d0   
8263      ff8ba31323d82367c785674b56546fed   
8264      ff9ac1dc643ec572ffc0949fff6748dc   
8265      ff9cb8a5a40d52ceaba34d3d56bc8d0f   
8266  ffa08430-dff2-11e5-8d98-4b3d9215ade1   
8267  ffa11d52-8d39-11e5-934c-a369c80822c2   
8268  ffb50cec-b800-11e2-b568-6917f6ac6d9d   
8269  ffc37f8c-8885-11e4-9534-f79a23c40e6c   
8270      ffd02191f62c5a1b606178dd63d7b47f   
8271      ffe956a44ff321b4d009d80be738c141   
8272      ffe9b46cf5ea35ef03d7065087771deb   
8273  ffee0314-3f0e-11e2-ae43-cf491b837f7b   
8274      fffa8e39802e6a74d8b6e9f432957dc0   

                                                    url  \
0     https://www.washingtonpost.com/news/wonk/wp/20...   
1     https://www.washingtonpost.com/local/one-injur...   
2     https://www.washingtonpost.com/news/wonk/wp/20...   
3     https://www.washingtonpost.com/opinions/why-po...   
4     https://www.washingtonpost.com/news/powerpost/...   
5     https://www.washingtonpost.com/news/wonk/wp/20...   
6     https://www.washingtonpost.com/lifestyle/magaz...   
7     https://www.washingtonpost.com/blogs/post-part...   
8     https://www.washingtonpost.com/news/dr-gridloc...   
9     https://www.washingtonpost.com/news/capital-we...   
10    https://www.washingtonpost.com/news/dr-gridloc...   
11    https://www.washingtonpost.com/news/to-your-he...   
12    https://www.washingtonpost.com/news/capital-we...   
13    https://www.washingtonpost.com/world/europe/eu...   
14    https://www.washingtonpost.com/news/the-fix/wp...   
15    https://www.washingtonpost.com/news/speaking-o...   
16    https://www.washingtonpost.com/news/to-your-he...   
17    https://www.washingtonpost.com/posteverything/...   
18    https://www.washingtonpost.com/news/capital-we...   
19     

now we have all question and answer reports from sql in df_answers and df_questions

In [7]:
print(df_answers.shape)
print(df_questions.shape)

(8275, 8)
(50, 8)


In [8]:
df_answers.describe()

id        url  \
count                                   8275       8275   
unique                                  8275       6961   
top     5617da28-f8f9-11e6-aa1e-5f735ee31334  duplicate   
freq                                       1        678   

                                                    title author  \
count                                                8275   8275   
unique                                               7559   1853   
top     What Hillary Clinton’s pneumonia diagnosis mea...          
freq                                                   72    458   

                 date               source articleType  \
count            8275                 8275        8275   
unique           7612                    1         148   
top     1473674685000  The Washington Post    Wonkblog   
freq               72                 8275         512   

                                                  content  
count                                                8275  
unique                                               7639  
top     On Labor Day, in the middle of a Cleveland cam...  
freq                                                   71

In [9]:
# merge the corresponding df
df_merged_questions = pd.merge(left=df_questions,right=questions_tup_df, left_on='id', right_on='id')
df_merged_questions

id  \
0       02e52bdba097c9df4cbae66e04f82542   
1   0e85b0c0-f7ef-11e4-9030-b4732caefe81   
2   11915bd8-7944-11e2-9c27-fdd594ea6286   
3       145b9a6caa16d931c108a89798e65e17   
4   159e6f9e-8e84-11e3-84e1-27626c5ef5fb   
5       1e03fecf4d33b7896203298ab3858156   
6       2a340b8573d498e261d6f2365b37f8eb   
7       2bea9433d4e1050c9c85175df466b3e2   
8       2e83ad87eb1bade22e6e96ece616c24f   
9   30a493b8-fb07-11e4-9ef4-1bb7ce3b3fb7   
10  30c00b60-13f6-11e3-b182-1b3bb2eb474c   
11      37a8e2283e4677b703f6464d0191a700   
12      3902c9005a0563742fc4acb2c011b164   
13  3c5be31e-24ab-11e5-b621-b55e495e9b78   
14  4192b016-8708-11e3-a5bd-844629433ba3   
15      43e9f3f12982c0e0bb15ad64b33a89c0   
16  474ae088-ab1e-11e4-9c91-e9d2f9fde644   
17      4989ebfeb752e6b317d1ef3997b21a01   
18  563fb77e-024f-11e6-9203-7b8670959b88   
19      56f0438ee0fb34c341ccf5af36de5175   
20  579e9ae8-6a2f-11e6-8225-fbb8a6fc65bc   
21      5c466d4a01492f1b5cc9758e19429a1f   
22      5ec40b6bc6c5f4487132da7be04fc914   
23      5f37aac53768e749b861028397eb6849   
24      6668d83480f5c58b54a90770835ac2d4   
25      6fdc62d37aaf685b809c501abe13c56c   
26      72e72b41097d53b627fd375dd2d3309b   
27      7ef8ce1720bf2f6b2065a97506ee89b4   
28      9171debc316e5e2782e0d2404ca7d09d   
29  985b90cc-7c98-11e3-93c1-0e888170b723   
30      988147454a2b8eafd1535cd673dd04ba   
31      9dd7b85cd1e3da1b5c8e79f32fec7177   
32  a1c41a70-35c7-11e3-8a0e-4e2cf80831fc   
33      a244d1e0cfd916a2af76b6a6c785b017   
34      a2744bb98e1968307548e4976232cf1c   
35  a36fa8a2-8962-11e6-bff0-d53f592f176e   
36      a79b1b7d8cc5273d4995fec5e122e44b   
37  b0235f56-1cce-11e4-ae54-0cfe1f974f8a   
38      b4c6361974466458bb721b9b1628220b   
39      bd1e6cc8d7525fec36a717be45638bf4   
40      c109cc839f2d2414251471c48ae5515c   
41      c3cea789141ef2ae856419e86e165e0c   
42      c6bf4a4bf542b7c67987c222d73def4b   
43  c8351276-76de-41f1-b294-4f3e5d373c8c   
44      cad56e871cd0bca6cc77e97ffe246258   
45      dcd1560bd13a0b665b95d3ba27cc960c   
46      defd7f4a85496d52a210938d58a7ae76   
47  e1336b8f-b0c2-4610-9a3c-ec85a546c9ad   
48      fc1ca759c9c433376e71884870d225ab   
49      fef0f232a9bd94bdb96bac48c7705503   

                                                  url  \
0   https://www.washingtonpost.com/news/speaking-o...   
1   https://www.washingtonpost.com/business/econom...   
2                                           duplicate   
3   https://www.washingtonpost.com/news/post-natio...   
4   https://www.washingtonpost.com/entertainment/m...   
5   https://www.washingtonpost.com/news/worldviews...   
6   https://www.washingtonpost.com/news/animalia/w...   
7   https://www.washingtonpost.com/news/morning-mi...   
8   https://www.washingtonpost.com/news/capital-we...   
9   https://www.washingtonpost.com/world/national-...   
10  https://www.washingtonpost.com/lifestyle/style...   
11  https://www.washingtonpost.com/news/wonk/wp/20...   
12  https://www.washingtonpost.com/news/worldviews...   
13  https://www.washingtonpost.com/world/europe/to...   
14  https://www.washingtonpost.com/politics/sugar-...   
15  https://www.washingtonpost.com/blogs/erik-wemp...   
16  https://www.washingtonpost.com/local/traffican...   
17  https://www.washingtonpost.com/news/post-natio...   
18  https://www.washingtonpost.com/local/traffican...   
19  https://www.washingtonpost.com/news/wonk/wp/20...   
20  https://www.washingtonpost.com/local/md-politi...   
21  https://www.washingtonpost.com/news/speaking-o...   
22  https://www.washingtonpost.com/news/capital-we...   
23  https://www.washingtonpost.com/news/wonk/wp/20...   
24  https://www.washingtonpost.com/news/worldviews...   
25  https://www.washingtonpost.com/blogs/govbeat/w...   
26  https://www.washingtonpost.com/news/energy-env...   
27  https://www.washingtonpost.com/news/dr-gridloc...   
28  https://www.washingtonpost.com/news/worldviews...   
29  https://www.washingtonpost.com/national/health...   
30  https://www.washington

In [10]:
df_merged_answers = pd.merge(left=df_answers,right=answers_tup_df, left_on='id', right_on='id')
df_merged_answers

id  \
0         00041887f95b15333ae39503610f2de1   
1     00054d36-2473-11e3-b3e9-d97fb087acd6   
2         000563252f7569174d0f1a2aff115854   
3     000da51c-f93f-11e2-a954-358d90d5d72d   
4         000f4c71d82856a1584c4a3c62c71454   
5         0013cc4281f4c81f611bb625031ac605   
6     0015851c-2170-11e3-966c-9c4293c47ebe   
7         003014d2b6c491c98fa759692e0c99da   
8         0032b35853f8288d99d4f92cb5794966   
9         00461b9f108425ff68f1dd910a6cf202   
10        005308753e2d6f8a71c8be31be210e28   
11        0058b29067c414f21b35c8f64cb49e64   
12        005a1f0c2064487a7f15443b2a5f349a   
13    0062ad00-f7ee-11e4-a47c-e56f4db884ed   
14        006f205a6298a2ff627b5236cb984463   
15        007506cd7a7fcc1e03aac3a8e93ba6db   
16        00796cc1f45678e59898bee51e7260ae   
17        007a0486368dd6aa54f2b62cd431c046   
18        007cf8e2f0867bf2db10a49162386e83   
19    007f0b42-0e28-11e2-bd1a-b868e65d57eb   
20        008026efacc89ec065782505f4da4472   
21        00809043cc3bbea3b4d584ce62d08b0b   
22        00876e25a722ae351db811349a8bd101   
23    0089c108-0d53-11e6-bfa1-4efa856caf2a   
24    008c981a-c2e0-11e4-ad5c-3b8ce89f1b89   
25    008c981a-c2e0-11e4-ad5c-3b8ce89f1b89   
26        0091c37f78cae2ab67cff31a52eb4f1f   
27    0091e9d4-c8f1-11e4-a199-6cb5e63819d2   
28    009aafb6-0283-11e6-8bb1-f124a43f84dc   
29        00a43ad2b87158552fa26eb0c9041df6   
...                                    ...   
8478  ff0ae020-4a6e-11e5-8ab4-c73967a143d3   
8479  ff13bb5c-aea5-11e3-a49e-76adc9210f19   
8480  ff1f5c2c-10ab-11e6-8967-7ac733c56f12   
8481  ff2d3122-21f2-11e2-bdfa-eebc58545bc7   
8482  ff308aa4-822f-11e3-bbe5-6a2a3141e3a9   
8483  ff3a25b0-0ba4-11e4-8341-b8072b1e7348   
8484      ff42f6cd0547632631044913241fec0d   
8485  ff47b876-714d-11e2-ac36-3d8d9dcaa2e2   
8486  ff49e708-5771-11e3-ba82-16ed03681809   
8487      ff4c7b7a86185b580010f20fad44a6d5   
8488      ff52231aec12758f729eeb4c56fc902a   
8489      ff56952e96544c44067ffc403fc753ba   
8490      ff59f055c6f05e9ce3aa91461f492bc2   
8491      ff63aded5d71af8cc553f22dca202370   
8492      ff63aded5d71af8cc553f22dca202370   
8493  ff6bd6e0-fe0e-11e2-9711-3708310f6f4d   
8494  ff7a152a-7303-11e7-9eac-d56bd5568db8   
8495      ff871ed540102bbc7ff03aba759f38d0   
8496      ff8ba31323d82367c785674b56546fed   
8497      ff9ac1dc643ec572ffc0949fff6748dc   
8498      ff9cb8a5a40d52ceaba34d3d56bc8d0f   
8499  ffa08430-dff2-11e5-8d98-4b3d9215ade1   
8500  ffa11d52-8d39-11e5-934c-a369c80822c2   
8501  ffb50cec-b800-11e2-b568-6917f6ac6d9d   
8502  ffc37f8c-8885-11e4-9534-f79a23c40e6c   
8503      ffd02191f62c5a1b606178dd63d7b47f   
8504      ffe956a44ff321b4d009d80be738c141   
8505      ffe9b46cf5ea35ef03d7065087771deb   
8506  ffee0314-3f0e-11e2-ae43-cf491b837f7b   
8507      fffa8e39802e6a74d8b6e9f432957dc0   

                                                    url  \
0     https://www.washingtonpost.com/news/wonk/wp/20...   
1     https://www.washingtonpost.com/local/one-injur...   
2     https://www.washingtonpost.com/news/wonk/wp/20...   
3     https://www.washingtonpost.com/opinions/why-po...   
4     https://www.washingtonpost.com/news/powerpost/...   
5     https://www.washingtonpost.com/news/wonk/wp/20...   
6     https://www.washingtonpost.com/lifestyle/magaz...   
7     https://www.washingtonpost.com/blogs/post-part...   
8     https://www.washingtonpost.com/news/dr-gridloc...   
9     https://www.washingtonpost.com/news/capital-we...   
10    https://www.washingtonpost.com/news/dr-gridloc...   
11    https://www.washingtonpost.com/news/to-your-he...   
12    https://www.washingtonpost.com/news/capital-we...   
13    https://www.washingtonpost.com/world/europe/eu...   
14    https://www.washingtonpost.com/news/the-fix/wp...   
15    https://www.washingtonpost.com/news/speaking-o...   
16    https://www.washingtonpost.com/news/to-your-he...   
17    https://www.washingtonpost.com/posteverything/...   
18    https://www.washingtonpost.com/news/capital-we...   
19     

### We are set. df_merged_questions and df_merged_answers are the only values we need to do exploration

#### Distribution of scores for all 8508 judgements

In [11]:
df_merged_answers["score"].value_counts()

0     6465
2     1189
4      584
8      164
16     106
Name: score, dtype: int64

In [12]:
a = df_merged_answers.drop_duplicates(subset="id")
a.shape

(8275, 10)

In [13]:
pd.concat([df_merged_answers, a, a]).drop_duplicates(keep=False) # ids that are duplicates

id  \
25    008c981a-c2e0-11e4-ad5c-3b8ce89f1b89   
124   036c654e-a313-11e3-8466-d34c451760b9   
147       041cba74b97a9b3c4b469cf161f83d31   
170       048e4b3282b7dca5184c2ebcc7748550   
187   04dddf78-e728-11e6-b82f-687d6e6a3e7c   
193       050307a650d0618078029ab90792ec69   
240       06228d0afcf0e8d36998c31d4f29ceaa   
249   0659042c-b3de-11e6-8616-52b15787add0   
251   0669e164-22cf-11e3-966c-9c4293c47ebe   
252   0669e164-22cf-11e3-966c-9c4293c47ebe   
265       06c36873cb6a42050c408e771699eb01   
277       071ad6f3cfba2273c7f8d5264c0a7583   
285   07469b62-9fee-11e3-b8d8-94577ff66b28   
301       07b35af4c149978c5a206bbdf3579b72   
352       0920797fca62455d8333d74c14b3e7eb   
388   09b3742e-fe9b-11e1-b153-218509a954e1   
451   0bcbb94e-0feb-11e3-bdf6-e4fc677d94a1   
492       0cb439142ab18761bc81b6dfad90ae85   
501       0ce7ed84c172864e53d7d8debad255b9   
565   0e95b2c0-a3b5-11e3-8466-d34c451760b9   
591       0f4f11f765cf38cb484ae7a9b622133a   
614       0fc67e79e3ebf15c610944c25b19abe4   
649       1098ceacafbdf041037b0416dccd7ffa   
655       10bbd6c3e72dd4f2b87c9c8e0e59495a   
722       128dae1b2ea002255a89ad7833b34185   
740       12ef93fa82d9af43ba8b4701ba64dff7   
745       131a8630a4d8294a14b133bd82464866   
811       14c70f00d6352c11ba01a46652365b58   
851   15a49a2c-b84f-11e4-a200-c008a01a6692   
859       15dc1e84815b86cef8db255fae8cc6f1   
...                                    ...   
7204      d7306d580d8711ec73e1b22a6579a4ba   
7299      da2f10b10fe6e5b71ba4d41ac51d46e7   
7414      dd9c4b8fbdfa2081b898a219a803bf05   
7482  df9a34da-a59a-11e4-a7c2-03d37af98440   
7500      e04fd58874a847df918bfcf2d654abce   
7520  e0c93358-887f-11e3-a5bd-844629433ba3   
7562  e1e7acc6-751a-11e1-8fa6-9fcb4e001234   
7592      e2c53c13997a2cda220116248a6f010a   
7615  e3afdb62-1f06-11e4-82f9-2cd6fa8da5c4   
7705  e6bc59ac-c69e-11e5-8965-0607e0e265ce   
7723      e70bcceb01ae9d6295cab72ff952f642   
7733      e7589bae8b831f26d608959819ae45c9   
7789      e900b10aa18398ecb99ec58136fca5fc   
7795      e922d350648be146c9ac683ec24742e3   
7811  e9ba25a0-efe8-11e3-914c-1fbd0614e2d4   
7838      ea6c9dafa7aa45e1bb10517e149459d0   
7865      eb56d6574d8eb4e1504e746c14181a18   
7895  ec608662-d53c-11e3-aae8-c2d44bd79778   
8113  f37bf3bc-b399-11e6-be1c-8cec35b1ad25   
8114  f37bf3bc-b399-11e6-be1c-8cec35b1ad25   
8157      f4c92a47497c69b8cf7ab1b3c7264076   
8160  f4d93e38-508e-11e4-babe-e91da079cb8a   
8192      f5fac61394ebb4df3f3c2945de559202   
8239      f7759c48bdc46a54875f43287fbf062f   
8350      fae7de2569233a17b6c7ecaade48e148   
8370      fb8f33de5fd6f7fb436ecb6c3d09458f   
8373  fb96aff2-168d-11e4-9e3b-7f2f110c6265   
8430      fd808a96ac064a15cc3fc37e09fc8b1e   
8471      fec7dda0e6ed2316040691b030817a4b   
8492      ff63aded5d71af8cc553f22dca202370   

                                                    url  \
25    https://www.washingtonpost.com/national/health...   
124   https://www.washingtonpost.com/national/health...   
147   https://www.washingtonpost.com/news/worldviews...   
170   https://www.washingtonpost.com/news/wonk/wp/20...   
187   https://www.washingtonpost.com/opinions/how-ma...   
193   https://www.washingtonpost.com/news/wonk/wp/20...   
240   https://www.washingtonpost.com/news/to-your-he...   
249   https://www.washingtonpost.com/opinions/fidel-...   
251   https://www.washingtonpost.com/local/obituarie...   
252   https://www.washingtonpost.com/local/obituarie...   
265   https://www.washingtonpost.com/news/morning-mi...   
277   https://www.washingtonpost.com/news/worldviews...   
285   https://www.washingtonpost.com/business/econom...   
301   https://www.washingtonpost.com/news/morning-mi...   
352   https://www.washingtonpost.com/news/morning-mi...   
388                                           duplicate   
451   https://www.washingtonpost.com/business/techno...   
492   https://www.washingtonpost.com/news/to-your-he...   
501   https://www.washingtonpost.com/news/the-watch/...   
565   h

In [14]:
# instance where a report occurs in two relevance judgements
df_merged_answers.loc[df_merged_answers["id"]=='048e4b3282b7dca5184c2ebcc7748550']

id  \
169  048e4b3282b7dca5184c2ebcc7748550   
170  048e4b3282b7dca5184c2ebcc7748550   

                                                   url  \
169  https://www.washingtonpost.com/news/wonk/wp/20...   
170  https://www.washingtonpost.com/news/wonk/wp/20...   

                                                 title           author  \
169  Hillary Clinton’s top goal as president could ...  Max Ehrenfreund   
170  Hillary Clinton’s top goal as president could ...  Max Ehrenfreund   

              date               source articleType  \
169  1437399593000  The Washington Post    Wonkblog   
170  1437399593000  The Washington Post    Wonkblog   

                                               content topic_no score  
169  Hillary Rodham Clinton says the "defining econ...      803     0  
170  Hillary Rodham Clinton says the "defining econ...      816     2

In [ ]:
df_merged_answers